File: orchalanguage/src/test/java/orcha/lang/compiler/referenceimpl/CompilerReferenceImplTestConfiguration.java
package orcha.lang.compiler.referenceimpl;

import orcha.lang.compiler.*;
import orcha.lang.compiler.syntax.SendInstruction;
import orcha.lang.compiler.syntax.WhenInstruction;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.SpringBootConfiguration;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.DependsOn;

@SpringBootConfiguration
@Configuration
public class CompilerReferenceImplTestConfiguration {

    @Bean(name = "whenInstruction")
    public WhenInstructionFactory whenInstructionFactory() {
        WhenInstructionFactory factory = new WhenInstructionFactory();
        return factory;
    }

    @Bean
    public WhenInstruction whenInstruction() throws Exception {
        return whenInstructionFactory().getObject();
    }

    @Bean(name = "sendInstruction")
    public SendInstructionFactory sendInstructionFactory() {
        SendInstructionFactory factory = new SendInstructionFactory();
        return factory;
    }

    @Bean
    public SendInstruction sendInstruction() throws Exception {
        return sendInstructionFactory().getObject();
    }

    @Value("${orcha.pathToBinaryCode:build/resources/main}")
    String pathToBinaryCode;

    @Bean
    Preprocessing preprocessingForTest() {
        return new PreprocessingImpl();
    }

    @Bean
    @DependsOn({"whenInstruction", "sendInstruction"})
    LexicalAnalysis lexicalAnalysisForTest() {
        return new LexicalAnalysisImpl();
    }

    @Bean
    SyntaxAnalysis syntaxAnalysisForTest() {
        return new SyntaxAnalysisImpl();
    }

    @Bean
    SemanticAnalysis semanticAnalysisForTest() {
        return new SemanticAnalysisImpl();
    }

}

File: orchalang-spring-integration-implementation/src/test/kotlin/orcha/lang/compiler/referenceimpl/CompilerReferenceImplTestConfiguration.kt
package orcha.lang.compiler.referenceimpl

import orcha.lang.compiler.LexicalAnalysis
import orcha.lang.compiler.LinkEditor
import orcha.lang.compiler.SemanticAnalysis
import orcha.lang.compiler.SyntaxAnalysis
import orcha.lang.compiler.referenceimpl.springIntegration.LinkEditorImpl
import orcha.lang.compiler.referenceimpl.springIntegration.SendInstructionFactory
import orcha.lang.compiler.referenceimpl.springIntegration.WhenInstructionFactory
import orcha.lang.compiler.syntax.SendInstruction
import orcha.lang.compiler.syntax.WhenInstruction
import orcha.lang.configuration.*
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.SpringBootConfiguration
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.DependsOn

@SpringBootConfiguration
@TestConfiguration
class CompilerReferenceImplTestConfiguration {

    //@Value("\${orcha.pathToBinaryCode:build/resources/main}")
    //internal var pathToBinaryCode: String? = null

    @Bean
    @DependsOn("whenInstruction", "sendInstruction")
    internal fun lexicalAnalysisForTest(): LexicalAnalysis {
        return LexicalAnalysisImpl()
    }

    @Bean(name = ["whenInstruction"])
    fun whenInstructionFactory(): WhenInstructionFactory {
        return WhenInstructionFactory()
    }

    @Bean
    @Throws(Exception::class)
    fun whenInstruction(): WhenInstruction? {
        return whenInstructionFactory().getObject()
    }

    @Bean(name = ["sendInstruction"])
    fun sendInstructionFactory(): SendInstructionFactory {
        return SendInstructionFactory()
    }

    @Bean
    @Throws(Exception::class)
    fun sendInstruction(): SendInstruction? {
        return sendInstructionFactory().getObject()
    }

    @Bean
    internal fun syntaxAnalysisForTest(): SyntaxAnalysis {
        return SyntaxAnalysisImpl()
    }

    @Bean
    internal fun semanticAnalysisForTest(): SemanticAnalysis {
        return SemanticAnalysisImpl()
    }

    @Bean
    fun travelAgency(): EventHandler {
        val eventHandler = EventHandler("travelAgency")
        val fileAdapter = InputFileAdapter("./files", "*.json")
        eventHandler.input = Input(fileAdapter, "java.lang.String")
        return eventHandler;
    }

    @Bean
    fun checkOrder(): Application {
        val application = Application("checkOrder", "Kotlin")
        val javaAdapter = JavaServiceAdapter("orcha.lang.compiler.referenceimpl.PreprocessingImpl", "process")
        application.input = Input(javaAdapter, "java.lang.String")
        application.output = Output(javaAdapter, "java.util.List<java.lang.String>")
        return application
    }

    @Bean
    fun customer(): EventHandler {
        val eventHandler = EventHandler("customer")
        val fileAdapter = OutputFileAdapter("data/output", "orchaCompiler.java")
        eventHandler.output = Output(fileAdapter, "application/java-archive")
        return eventHandler
    }

    @Bean
    internal fun linkEditorForTest(): LinkEditor {
        return LinkEditorImpl()
    }
}
